### ISSUE TITLE
GetDocuments: `whereField` creates a memory leak. 

### ISSUE THREAD

**Issue by jorgefrazaocosta** [firebase/firebase-ios-sdk#12613]
### Description

Using Leaks, it's possible to see that when we use the `whereField`, there is a memory leak being created. We saw it in several and different calls, all of them the Leaks tool points to the `whereField`.

### Reproducing the issue

1. Adds Firebase Firestore to the project
2. Read documents from a Collection and filter some results using the `whereField` method from FIRQuery
3. Launch App
4. Observe memory leak using Leaks tool

### Firebase SDK Version

10.23.0

### Xcode Version

15.3

### Installation Method

Swift Package Manager

### Firebase Product(s)

AB Testing, Analytics, Authentication, Crashlytics, Firestore, Remote Config

### Targeted Platforms

iOS

### Relevant Log Output

<img width="1474" alt="Screenshot 2024-03-22 at 15 03 40" src="https://github.com/firebase/firebase-ios-sdk/assets/2707674/205c9d52-d76a-4cc0-8e0e-150d3e194cee">

_No response_

**Comment by tom-andersen** [comment]
Thank you for the report. I will look into this as soon as possible.

**Comment by tom-andersen** [comment]
@jorgefrazaocosta Does this happen in version 10.15.0? We suspect maybe this might have been introduced in 10.16.0.

**Comment by jorgefrazaocosta** [comment]
@tom-andersen Indeed. It doesn't happen in version 10.15.0; I can see it using 10.16.0.

**Comment by tom-andersen** [comment]
I am assigning this issue to @milaGGL who is more familiar with this part of the code.

**Comment by SwiftUIBarcode** [comment]
Hello any updates on this I am still having this issue @milaGGL

**Comment by milaGGL** [comment]
Hi @SwiftUIBarcode, there is a fix (https://github.com/firebase/firebase-ios-sdk/pull/12972) created and it's still in progress. Unfortunately, I can't give you an ETA for when it will be finished. I will keep this thread updated once there is a progress.

**Comment by SwiftUIBarcode** [comment]
@milaGGL  can confirm what @jorgefrazaocosta  said a long time ago. Tested it again in the new year 2025, 10.15.0 doesn't have the memory leak .wherefield parse , but the 10.16.0 onwards does. Is this not a really important issue I feel like .wherefield is such a common use case in firebase iOS SDK that I'm really surprised no one else is even making issue threads about this anymore as it is still present in the code . I have 11.6.0 and I am getting these leaks. I know you said you will update the thread but I'm just confused about this.!![Image](https://github.com/user-attachments/assets/500bb4d0-74c0-4b95-9743-04475f97b2bd)
![Image](https://github.com/user-attachments/assets/6ff2b20a-600a-4f48-949c-6bacb5a8d691)
